test/test_s3.py

```diff
@@ -41,6 +45,53 @@
 MB = 1024 ** 2
 GB = 1024 ** 3
 
+cross_process_lock_name = "instance_lock_test"
+
+def cross_proc_task():
+    try:
+        lock = crt_instance_lock_acquire(cross_process_lock_name)
+        lock.release()
+        exit(0)
+    except RuntimeError as e:
+        exit(-1)      
+
+class InstanceLockTest(NativeResourceTest):
+    def setUp(self):
+        self.nonce = time.time()
+        super().setUp()
+
+    def test_with_statement(self):
+        nonce_str = 'lock_a_{}'.format(self.nonce)
+        with crt_instance_lock_acquire(nonce_str) as lock:
+            try:
+                new_lock = crt_instance_lock_acquire(nonce_str)
+                self.fail("Acquiring a lock by the same nonce should fail when it's already held")
+            except RuntimeError as e:
```

Comment:
Does the CRT have any formalized error hierarchy in Python? I haven't dug too deeply yet. My only concern here is RuntimeError is very generic so it will be difficult to do control flow handling with this without message introspection.

Reply:
@graebm  any thoughts here?

Reply:
It's not great. It's the same problem we have in every language binding. We didn't code-gen error classes, so catching specific CRT errors means users must catch a RuntimeError and introspect it.

In this specific case, we could dance around the issue and just return None or False if the C function returns NULL, since we do often expect it to fail 🤷‍♀️

Reply:
I'm not happy about this. I really want to do a mildly-breaking-change to all our bindings and address this issue. But it's non-trivial :|